### PR TITLE
Updated build to use latest version of lfc

### PR DESCRIPTION
**Pull request by gundralaa**
uses nightly build

### CODE REVIEW

**Comment by gundralaa** [comment]
modes.h has a conflict in the nrf52 dependencies

**Comment by lhstrh** [comment]
> modes.h has a conflict in the nrf52 dependencies

What is the conflict?

File: platform/Makefile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Main source and header files
+APP_HEADER_PATHS += .
+APP_SOURCE_PATHS += .
+APP_SOURCES += $(notdir $(wildcard ./*.c))
+
+# Include board Makefile (if any)
+include ../../boards/lf_buckler_revC/Board.mk
+
+# Include main Makefile
+include ../../buckler/software/nrf52x-base/make/AppMakefile.mk
```
Comment (erlingrj) [inline platform/Makefile]: These relative includes are not working for all cases. E.g. if you try to compile test/TestAction.lf

File: platform/lf_types.h [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+ *
+ * Type definitions that are widely used across different parts of the runtime.
+ */
+
+#ifndef TYPES_H
+#define TYPES_H
+
+#include <stdbool.h>
+
+#include "modal_models/modes.h" // Modal model support
```
Comment (erlingrj) [inline platform/lf_types.h]: Could you submit a PR to reactor-c with this change?

File: scripts/build_nrf_unix.sh [hunk 3]
```diff
@@ -7,55 +7,16 @@ PROJECT_ROOT=$LF_BIN_DIRECTORY/..
...
-cp $PROJECT_ROOT/platform/lf_nrf52_support.c $LF_SOURCE_GEN_DIRECTORY/include/core/platform/
-cp $PROJECT_ROOT/platform/lf_nrf52_support.h $LF_SOURCE_GEN_DIRECTORY/include/core/platform/
-cp $PROJECT_ROOT/platform/platform.h $LF_SOURCE_GEN_DIRECTORY/include/core/
-cp $PROJECT_ROOT/platform/reactor.c $LF_SOURCE_GEN_DIRECTORY/include/core/
-cp $PROJECT_ROOT/platform/reactor_common.c $LF_SOURCE_GEN_DIRECTORY/include/core/
-
 # Copy the lib directory.
 cp -r $PROJECT_ROOT/lib/* $LF_SOURCE_GEN_DIRECTORY/lib
+# Copy makefile
+cp $PROJECT_ROOT/platform/Makefile $LF_SOURCE_GEN_DIRECTORY/
```
Comment (erlingrj) [inline scripts/build_nrf_unix.sh]: The convenient thing with having a script that generated the makefile is that we could easily find the root of the project and thus get a correct path to the NRF SDKs. An alternative is to pass the project root as an argument to make.  A better alternative might be for this script to open Makefil and add add a line to the top e.g.
```
PROJECT_ROOT=$LF_BIN_DIRECTORY/..
```

Then save it into the new location.

**Review by erlingrj — COMMENTED** [review]
Well done

File: platform/lf_types.h (hunk 2 above)
Comment (lhstrh) [reply platform/lf_types.h]: Tagging @gundralaa for visibility.

**Comment by erlingrj** [comment]
Do you have access to a NRF52, @gundralaa? If so you can test that this is working by running "make test" it should compile, flash and parse the output from a couple of simple test LF programs. Also, it would be better to make the minor change to `lf_types.h` directly in reactor-c. We will fix the include-path problem soon

File: scripts/build_nrf_unix.sh [hunk 4]
```diff
@@ -7,55 +7,20 @@ PROJECT_ROOT=$LF_BIN_DIRECTORY/..
...
-NRF_IC = nrf52832
-SDK_VERSION = 15
-SOFTDEVICE_MODEL = s132
-
-# LF Sources and Headers
-APP_SOURCES += $(notdir lf_nrf52_support.c)
-APP_SOURCES += $(notdir $(wildcard ./lib/*.c))
-APP_SOURCE_PATHS += ./core/platform/
-APP_SOURCE_PATHS += ./lib/
+# TODO Push into reactor-c
```
Comment (lhstrh) [inline scripts/build_nrf_unix.sh]:
```suggestion
```

File: scripts/build_nrf_unix.sh [hunk 5]
```diff
@@ -7,55 +7,20 @@ PROJECT_ROOT=$LF_BIN_DIRECTORY/..
...
+rm $LF_SOURCE_GEN_DIRECTORY/lib/util.c
```
Comment (lhstrh) [inline scripts/build_nrf_unix.sh]:
```suggestion
```
First merge https://github.com/lf-lang/reactor-c/pull/148

File: platform/Makefile [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# TODO
+# lf_types.h needs header change from "modes.h" to "modal_models/modes.h" due to conflicting header in sdk
+# find ways to keep priority
+# delete deprecated util.c in lib
```
Comment (lhstrh) [inline platform/Makefile]:
```suggestion
```

File: platform/Makefile [hunk 7]
```diff
@@ -0,0 +1,62 @@
+# TODO
+# lf_types.h needs header change from "modes.h" to "modal_models/modes.h" due to conflicting header in sdk
```
Comment (lhstrh) [inline platform/Makefile]:
```suggestion
```

File: platform/Makefile [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+# find ways to keep priority
```
Comment (lhstrh) [inline platform/Makefile]: What does this comment refer to?

File: platform/Makefile [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+APP_SOURCE_PATHS += .
+APP_SOURCES += $(notdir $(wildcard ./*.c))
+
+NRF_BASE_DIR = $(PROJECT_ROOT)/buckler/software/nrf52x-base/
+
+# Include board Makefile (if any)
+include $(PROJECT_ROOT)/boards/lf_buckler_revC/Board.mk
+
+# Include main Makefile
+include $(PROJECT_ROOT)/buckler/software/nrf52x-base/make/AppMakefile.mk
```
Comment (lhstrh) [inline platform/Makefile]:
```suggestion
include $(PROJECT_ROOT)/buckler/software/nrf52x-base/make/AppMakefile.mk

```

**Review by lhstrh — CHANGES_REQUESTED** [review]
This looks great! Thanks, @gundralaa! I'll test it on the robot on Monday and approve this unless I run into any issues.

File: scripts/build_nrf_unix.sh [hunk 10]
```diff
@@ -7,55 +7,20 @@ PROJECT_ROOT=$LF_BIN_DIRECTORY/..
...
-# Copy platform into /include/core
-# TODO: Why are there two generated core dirs
-cp $PROJECT_ROOT/platform/lf_nrf52_support.c $LF_SOURCE_GEN_DIRECTORY/include/core/platform/
-cp $PROJECT_ROOT/platform/lf_nrf52_support.h $LF_SOURCE_GEN_DIRECTORY/include/core/platform/
-cp $PROJECT_ROOT/platform/platform.h $LF_SOURCE_GEN_DIRECTORY/include/core/
-cp $PROJECT_ROOT/platform/reactor.c $LF_SOURCE_GEN_DIRECTORY/include/core/
-cp $PROJECT_ROOT/platform/reactor_common.c $LF_SOURCE_GEN_DIRECTORY/include/core/
-
 # Copy the lib directory.
 cp -r $PROJECT_ROOT/lib/* $LF_SOURCE_GEN_DIRECTORY/lib
```
Comment (erlingrj) [inline scripts/build_nrf_unix.sh:11]: Is this still necessary?

Comment (gundralaa) [reply scripts/build_nrf_unix.sh:11]: This copies some buckler specific lib dependencies like romi.c to the src-gen so still think its required.

**Review by lhstrh — APPROVED** [review]

**Comment by lhstrh** [comment]
Confirming that the tests all pass.

**Pull request merged**